### PR TITLE
[API] GetRelease by tag only return release

### PR DESCRIPTION
**Pull request by cameronbraid**
get release by tag should filter out tag releases to be consistent with list releases and get by id

There is a inconsistency between the 'list releases for repo' and 'get release by tag'.

The former filters out any release that is_tag=true where as the later includes them. 

So there is no consistent way to identify if a release exists for a given tag name.

If I iterate the 'list releases for repo' results it wont be there.  If I use 'get release by tag' it will be there but won't have any fields like title/notes/commitssh

### CODE REVIEW

File: routers/api/v1/repo/release_tags.go [hunk 1]
```diff
@@ -55,6 +55,12 @@ func GetReleaseTag(ctx *context.APIContext) {
 		return
 	}
 
+	if err != nil && models.IsErrReleaseNotExist(err) ||
+		release.IsTag || release.RepoID != ctx.Repo.Repository.ID {
```
Comment (lafriks) [inline routers/api/v1/repo/release_tags.go]: I don't think it's needed to check repoid

Comment (cameronbraid) [reply routers/api/v1/repo/release_tags.go]: Sure, i've removed that test

**Review by lafriks — APPROVED** [review]

**Comment by cameronbraid** [comment]
Can I also backport this to release/v1.13 ?

**Comment by CirnoT** [comment]
Makes it even more inconsistent due to `DELETE` being used to remove tag.

Your solution is sound however it would require:
- Removing `DELETE` on `​/repos​/{owner}​/{repo}​/releases​/tags​/{tag}` (having endpoint that removes tag grouped under release seems counter-intuitive anyway, release is not a tag nor can release have more than one tag)
- Adding `GET` for `/repos/{owner}/{repo}/tags/{tag}` to fetch specific tag
- Adding `DELETE` for `/repos​/{owner}​/{repo}​/tags/{tag}` to remove specific tag

**Comment by cameronbraid** [comment]
Not sure what you want conceptually for how gitea wants release/tags to function

From my limited understanding, a tag can exist without a release - however there is still a row in the release table for said tag

A release can be created for a tag which upgrades the row in the release table to be is_tag = false and fills on extra information 

I assumes that the `GET`/`DELETE` on `/repos​/{owner}​/{repo}​/releases​/tags​/{tag}` were alternative endpoints for `GET`/`DELETE` on `/repos/{owner}/{repo}/releases/{id}` but doing the lookups buy tag name instead of release id

**Review by a1012112796 — APPROVED** [review]

**Comment by CirnoT** [comment]
>I assumes that the GET/DELETE on /repos​/{owner}​/{repo}​/releases​/tags​/{tag} were alternative endpoints for GET/DELETE on /repos/{owner}/{repo}/releases/{id} but doing the lookups buy tag name instead of release id

No, the `DELETE` endpoint works only if the release is `is_tag=true` and removes is used to remove tag via git itself:
https://github.com/go-gitea/gitea/blob/8c086bae6e4a041234ba0f0463a392952386cc18/services/release/release.go#L138

By changing how `GET` endpoint works, you make it inconsistent with how `DELETE` behaves (note it can't be used to delete release!)
https://github.com/go-gitea/gitea/blob/76da070acce383966501990fe0d3b649e61ab5db/routers/api/v1/repo/release_tags.go#L111-L113

**Review by 6543 — CHANGES_REQUESTED** [review]
this is a breaking change!
and should at first be considdered carefully!

**Comment by 6543** [comment]
was added at #12932 related: #13358

File: routers/api/v1/repo/release_tags.go [hunk 2]
```diff
@@ -55,6 +55,11 @@ func GetReleaseTag(ctx *context.APIContext) {
 		return
 	}
 
+	if err != nil && models.IsErrReleaseNotExist(err) || release.IsTag {
```
Comment (6543) [inline routers/api/v1/repo/release_tags.go]: this is already checked!!! (at L48)

the only new thing is the `release.IsTag` check ... witch will break things

If we are going this way, we have to invent a new endpoint for git tags only

**Comment by 6543** [comment]
Summary of my review:

the  #13358 is not jet been released ....

so we could make `DELETE /repos/{owner}/{repo}/releases/tags/{tag}` to be used as "delete release by tag"
we then have to add a delete git tag  endpoint

I would suggest `DELETE /repos/{owner}/{repo}/tags/{tag}`
and add the check suggested by this pull ...

**Comment by johanvdw** [comment]
It seems the github api is also filtering out tags without release:
https://api.github.com/repos/go-gitea/gitea/releases/tags/v0.9.9
vs
https://api.github.com/repos/go-gitea/gitea/releases/tags/v1.0.0

**Comment by 6543** [comment]
I'll adjust this pull & send a follow up ...

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Pull request merged**